### ISSUE TITLE
Check how muse behaves when we increase the number of latent variables

### ISSUE THREAD

**Issue by Justinezgh** [Justinezgh/muse#2]
For these 3 experiments I use the following MUSE algorithm configuration:

```python
theta_start = jnp.array([0.3, 0.8])

prob.solve(
    result=result,
    α=0.2,
    θ_start=theta_start,
    θ_rtol=0,
    z_tol=1e-2,
    progress=True,
    maxsteps=100,
    nsims=100,
    rng=jax.random.PRNGKey(1)
)

```

simulator configuration: 

``` python 
nb_latent_variables = 100

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228060828-60b8fccd-3e78-49ec-9d60-4874ad94848f.png)
Convergence: 
![image](https://user-images.githubusercontent.com/72011736/228207805-366c9edd-127d-4811-a1cd-f2bdcb424b5e.png)

Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228061556-7fa8d2fe-5d2b-4bd8-a596-b6d848651b4c.png)

____________________________________________________________________________________________________

simulator configuration: 

``` python 
nb_latent_variables = 200

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228060963-3abed07d-6ae1-488c-b662-9f4e3378a0d7.png)

Convergence: 
![image](https://user-images.githubusercontent.com/72011736/228207934-ecc1d5ef-3825-4553-bb09-64d99a761de6.png)

Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228061691-b7a9d0e8-991c-4451-8017-8ef4e13b3c12.png)

________________________________________________________________________________________________

simulator configuration: 

``` python 
nb_latent_variables = 250

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228061080-fbd9202b-e0ca-4cb9-911a-8580d2a3f427.png)

Convergence: 
![image](https://user-images.githubusercontent.com/72011736/228208000-3825f026-ec9c-4414-be1e-bf10efe36c97.png)

Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228061912-a23835cc-19a3-44af-8a55-079251fd1150.png)

**Comment by Justinezgh** [comment]
simulator configuration: 

``` python 
nb_latent_variables = 200

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
)
```

![image](https://user-images.githubusercontent.com/72011736/226906969-8b0d40c5-843c-4bae-bbb0-3c9494fc7e77.png)

**Comment by Justinezgh** [comment]
**To do**

- [x]  change the starting point theta (was set to the fiducial params) to check if it can still converge

> Update: I changed it to [0.33, 0.83] 
![image](https://user-images.githubusercontent.com/72011736/228209491-431a1a84-dcfc-4b65-a9c5-fd9538afbef7.png)
the results are below 


- [ ] reduce the step size to make the score closer to zero

**Comment by Justinezgh** [comment]
For these 3 experiments I use the following MUSE algorithm configuration:

```python
theta_start = jnp.array([0.33, 0.83])

prob.solve(
    result=result,
    α=0.2,
    θ_start=theta_start,
    θ_rtol=0,
    z_tol=1e-2,
    progress=True,
    maxsteps=100,
    nsims=100,
    rng=jax.random.PRNGKey(1)
)

```


simulator configuration: 

``` python 
nb_latent_variables = 100

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228204063-0da6f986-c6e6-4645-af34-b74efb17ed9e.png)
Convergence:
![image](https://user-images.githubusercontent.com/72011736/228204475-0edf5fed-66d3-4a3e-879e-ebd6b339201c.png)
Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228204594-fa1d57b3-2e26-4076-8243-d8c7ad68a2ed.png)



____________________________________________________________________________________________________

simulator configuration: 

``` python 
nb_latent_variables = 200

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228204119-a43d197d-8570-4132-ab6c-e2f35ade5d58.png)

Convergence:
![image](https://user-images.githubusercontent.com/72011736/228205007-5adc97e2-2d09-4cbe-8e0d-92c4ba8f1286.png)
Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228205055-787a3fed-cca2-43a5-9cfd-519384bbbc9d.png)



________________________________________________________________________________________________


simulator configuration: 

``` python 
nb_latent_variables = 250

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

![image](https://user-images.githubusercontent.com/72011736/228204173-b9914bd4-6bc8-405a-9667-bb717a1e527e.png)

Convergence: 
![image](https://user-images.githubusercontent.com/72011736/228205204-5fbb8ec4-e8c4-4a21-8924-817fda2c3e60.png)

Loss of the last 90 iterations:
![image](https://user-images.githubusercontent.com/72011736/228205258-98fd503a-626c-4989-9c7f-a40f22efe582.png)

________________________________________________________________________________________________


simulator configuration: 

``` python 
nb_latent_variables = 500

model = partial(
    lensingLogNormal, 
    N=nb_latent_variables,
    map_size=5,
    gal_per_arcmin2=30,
    sigma_e=0.2,
    model_type='lognormal',
   non_gaussianity=0.03
)
```

(I don't have the full field contour for this one because 500 * 500 was too much for my GPUs memory)
![image](https://user-images.githubusercontent.com/72011736/228241292-90f3d591-1c3f-466d-9dea-0c654c0b4cc6.png)


Convergence: 
![image](https://user-images.githubusercontent.com/72011736/228241048-e0ee1f38-98d6-4614-bfa8-df8556d2fcc9.png)


Loss:
![image](https://user-images.githubusercontent.com/72011736/228241111-3a58723e-f8a1-4522-85fb-011b5e706716.png)

**Comment by Justinezgh** [comment]
> **To do**
> 
> * [x]   change the starting point theta (was set to the fiducial params) to check if it can still converge
> 
> > Update: I changed it to [0.33, 0.83]
> > ![image](https://user-images.githubusercontent.com/72011736/228209491-431a1a84-dcfc-4b65-a9c5-fd9538afbef7.png)
> > the results are below
> 
> * [ ]  reduce the step size to make the score closer to zero

- [ ] understand why when the starting point for the gradient descent is far away the covariance matrix is better approximated.
